core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java

```diff
@@ -562,4 +562,18 @@ public void testPeakMemoryUsed() throws Exception {
     }
   }
 
+  @Test
+  public void writeEmptyIteratorNotCreateEmptySpillFile() throws Exception {
+    final UnsafeShuffleWriter<Object, Object> writer = createWriter(true);
+    writer.write(Iterators.emptyIterator());
+    final Option<MapStatus> mapStatus = writer.stop(true);
+    assertTrue(mapStatus.isDefined());
+    assertTrue(mergedOutputFile.exists());
+    assertEquals(0, spillFilesCreated.size());
```

Comment:
It seems that we can move this line to `writeEmptyIterator()` instead of making a new test case.

Reply:
Test writeEmptyIterator() has create spill file although write  empty iterator,but the writeEmptyIteratorNotCreateEmptySpillFile test not create spill file while there has not records in memroy

Reply:
Do you mean that `writeEmptyIterator` will fail if we add this line `assertEquals(0, spillFilesCreated.size());` there?

Reply:
I mean that before add code "if (sortedRecords.hasNext()) { return }" it will fail. now add assertEquals(0, spillFilesCreated.size()) to writeEmptyIterator seems good.

Reply:
Ya. That's the same what I observed. There is no need to keep both test cases because there is no code path like that.

Reply:
That's it.